### PR TITLE
More dupe word typos

### PR DESCRIPTION
**Pull request by Rageking8**
I only picked those changes (from the regex search) that I am pretty certain doesn't change meaning and is just a typo fix. Do correct me if any fix is undesirable and I can revert those. Thanks.

### CODE REVIEW

File: compiler/rustc_lint_defs/src/builtin.rs [hunk 1]
```diff
@@ -3937,7 +3937,7 @@ declare_lint! {
     ///
     /// The compiler disables the automatic implementation if an explicit one
     /// exists for given type constructor. The exact rules governing this
-    /// are currently unsound and quite subtle and and will be modified in the future.
+    /// are currently unsound, quite subtle and will be modified in the future.
```
Comment (saethlin) [inline compiler/rustc_lint_defs/src/builtin.rs:3940]:
```suggestion
    /// are currently unsound, quite subtle, and will be modified in the future.
```

**Comment by saethlin** [comment]
I feel like there's some kind of study one could do on the kinds of spelling and grammar mistakes people often make, based on how many mistakes you've found here.

Thanks for finding these :heart:

**Comment by Rageking8** [comment]
> I feel like there's some kind of study one could do on the kinds of spelling and grammar mistakes people often make, based on how many mistakes you've found here.
> 
> Thanks for finding these :heart: 

Haha, there are at least 20-30 more in submodules which are not covered in this PR, together with a bunch that I am not sure about. I think the reason why so many duplicate words typo slip through reviews is because when proof reading the brain kinda just "filters" out the extra word. Leading to this many being found.

**Pull request merged**